### ISSUE TITLE
NETSCRIPT: level parameter of getActionRepGain is not really optional

### ISSUE THREAD

**Issue by bupjae** [bitburner-official/bitburner-src#257]
https://github.com/bitburner-official/bitburner-src/blob/303c54c85c7fddbeee312d224c4083ab1e4264de/src/NetscriptFunctions/Bladeburner.ts#L141-L155

On line 148, it seems the intention is make level parameter optional; if level parameter is not given, use current level of given action.

However, on line 144, if level parameter is not given properly, runtime error is thrown.

**Comment by Snarling** [comment]
Yeah that's my bad, pretty sure I messed that up. Checking the documentation verifies that level was supposed to be optional.

Moving the type verification for level after retrieving the action means we can select the correct default level, and we can also make rewardMultiplier a constant instead of needing to define it multiple ways based on a conditional.
```js
   //...
   const action = getBladeburnerActionObject(ctx, type, name); 
   const level = _level === undefined ? action.level : helpers.number(ctx, "level", _level); 
   const rewardMultiplier = Math.pow(action.rewardFac, level - 1); 
   //...
```

**Comment by Snarling** [comment]
Fixed in 80d751e
